src/scanner.c

```diff
@@ -382,6 +385,7 @@ UInt OpenInputStream (
     }
```

Comment:
That is not correct: TLS(Input) will be 0 the first time we invoke `OpenInput`, and we then need to skip the code inside the `if`.